### PR TITLE
Fix Part of #4938: Introduce Onboarding Intro screen

### PR DESCRIPTION
**Pull request by adhiamboperes**
## Explanation
Fix Part of https://github.com/oppia/oppia-android/issues/4938: New screen to explain to the learner the purpose of the app.

The new learner's name is added as a greeting, passed via the intent.

Tests and placeholder tests have been added to verify that all views are displayed and navigation works as expected.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
||Portrait|Landscape|
|---|---|---|
|Mobile Light Mode|Screenshot_1718117878|Screenshot_1718117885|
|Mobile Dark Mode|Screenshot_1718117908|Screenshot_1718117912|
|Tablet Light Mode|Screenshot_1718118054|Screenshot_1718118045|Screenshot_1718118119|
|Tablet Dark Mode|Screenshot_1718118119|Screenshot_1718118077|

### CODE REVIEW

File: app/src/main/java/org/oppia/android/app/onboarding/IntroFragment.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    super.onAttach(context)
+    (fragmentComponent as FragmentComponentImpl).inject(this)
+  }
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    val profileNickname = arguments!!.getStringFromBundle(PROFILE_NICKNAME_ARGUMENT_KEY)!!
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/onboarding/IntroFragment.kt]: We should use ``?.`` to access the bundle, and then a ``checkNotNull()`` as we do elsewhere.

Or, use protos for better type safety & field presence assurance.

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/IntroFragmentTest.kt [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+  fun testFragment_portraitMode_stepCountText_isDisplayed() {
+    launchOnboardingLearnerIntroActivity().use {
+      onView(withId(R.id.onboarding_steps_count))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.onboarding_steps_count))
+        .check(matches(withText(R.string.onboarding_step_count_four)))
+    }
+  }
+
+  @RunOn(TestPlatform.ESPRESSO) // Testing lifecycle fails on Robolectric.
```
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/onboarding/IntroFragmentTest.kt]: Ditto for these tests (can be made to work on Robolectric).

File: model/src/main/proto/arguments.proto [hunk 3]
```diff
@@ -326,3 +326,8 @@ message SurveyActivityParams {
   // The ID of the triggering exploration.
   string exploration_id = 3;
 }
+
+message IntroActivityParams {
```
Comment (BenHenning) [inline model/src/main/proto/arguments.proto:840]: Nit: add message doc comment for consistency.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @adhiamboperes! Just had a few comments--PTAL.

File: model/src/main/proto/arguments.proto (hunk 3 above)
Comment (adhiamboperes) [reply model/src/main/proto/arguments.proto:840]: Done

File: app/src/sharedTest/java/org/oppia/android/app/onboarding/IntroFragmentTest.kt (hunk 2 above)
Comment (adhiamboperes) [reply app/src/sharedTest/java/org/oppia/android/app/onboarding/IntroFragmentTest.kt]: Done

File: app/src/main/java/org/oppia/android/app/onboarding/IntroFragment.kt (hunk 1 above)
Comment (adhiamboperes) [reply app/src/main/java/org/oppia/android/app/onboarding/IntroFragment.kt]: Added the checkNotNull() option.

**Comment by adhiamboperes** [comment]
Thanks @BenHenning. PTAL at the revisions.

File: app/src/main/java/org/oppia/android/app/onboarding/IntroFragment.kt [hunk 4]
```diff
@@ -25,7 +25,10 @@ class IntroFragment : InjectableFragment() {
     container: ViewGroup?,
     savedInstanceState: Bundle?
   ): View? {
-    val profileNickname = arguments!!.getStringFromBundle(PROFILE_NICKNAME_ARGUMENT_KEY)!!
+    val profileNickname =
+      checkNotNull(arguments?.getStringFromBundle(PROFILE_NICKNAME_ARGUMENT_KEY)) {
+        "Expected profileNickname to be included in the arguments for IntroFragment"
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/onboarding/IntroFragment.kt]:
```suggestion
        "Expected profileNickname to be included in the arguments for IntroFragment."
```

**Review by BenHenning — APPROVED** [review]
Thanks @adhiamboperes! LGTM. I had just one nit--feel free to resolve it once addressed.

File: app/src/main/java/org/oppia/android/app/onboarding/IntroFragment.kt (hunk 4 above)
Comment (adhiamboperes) [reply app/src/main/java/org/oppia/android/app/onboarding/IntroFragment.kt]: Addressed comment. Resolving.

**Comment by adhiamboperes** [comment]
No pending comments, resolved conflicts and enabling auto-merge.

**Pull request merged**